ci-operator/step-registry/cucushift/upgrade/cpou/set-edge/4jumps/cucushift-upgrade-cpou-set-edge-4jumps-ref.yaml

```diff
@@ -0,0 +1,16 @@
+ref:
+  as: cucushift-upgrade-cpou-set-edge-4jumps
```

Comment:
Is this new step only used for 3 hops cpou jobs(4.14-4.17, 4.15-4.18)? how about 2 hops(for example, 4.16-4.18)?

Reply:
That will be done in another PR, with another step.

Reply:
hmm, I got u. My concern is that if we will add several steps to support 2 hops/3 hops and N hops in chain upgrade?

Reply:
Currently, I only have plan for 2 hops / 3 hops (because there are multiple tests/profiles on it).   For N hops, e.g, for 4.1->4.18, we don't change it very often, also we only have one or two profile on it. I don't think it can save us too much effort based on what I know currently.  We can talk/decide on the N hops later.

Reply:
> For N hops, e.g, for 4.1->4.18, we don't change it very often, also we only have one or two profile on it.

Seems not. It has more workloads in chain upgrade because both hops and profiles are more than cpou. take 4.17 for example, there are N paths(4.1/4.2/4.3/.../4.15 to 4.17, which means 15 paths) * M profiles(EOL has only one profile, but others has a maximum of 9 profiles) 

```
$ grep -rh "^- as:" *4.17-upgrade-from-stable-4.14*
- as: ibmcloud-ipi-ovn-ipsec-f28
- as: nutanix-ipi-proxy-fips-f28
- as: vsphere-ipi-ovn-dualstack-privmaryv6-f28
- as: baremetal-upi-ovn-ipv4-f28
- as: aws-ipi-proxy-sts-arm-f28
- as: azure-ipi-basecap-vset-additionalcap-arm-f28
- as: gcp-ipi-sdn-arm-f28
- as: aws-ipi-proxy-sts-arm-f28

$ grep -rh "^- as:" *4.17-upgrade-from-stable-4.13*
- as: baremetalds-ipi-ovn-ipv6-fips-f28
- as: gcp-upi-private-xpn-sdn-f28
- as: vsphere-ipi-ovn-dualstack-f28
- as: nutanix-ipi-proxy-fips-f28
- as: gcp-upi-private-xpn-sdn-f28
- as: baremetal-upi-ovn-ipv4-f28
- as: aws-ipi-proxy-sts-arm-f28
- as: aws-upi-basecap-none-arm-f28
- as: azure-ipi-ovn-ipsec-azurefile-csi-arm-f28
```
> We can talk/decide on the N hops later.

Fine from my side to take it into consideration later.

Reply:
> take 4.17 for example, there are N paths(4.1/4.2/4.3/.../4.15 to 4.17, which means 15 paths) * M profiles(EOL has only one profile, but others has a maximum of 9 profiles)

That's a different thing.  4.1->4.17 and 4.2->4.17 are different, because the hops are different. When the hops are different, a new step is needed (because different hops requires different upgrade edges, thus the dependencies numbers and shell commands to write the upgrade edge are different). That means we will need a new step (for each of 4.1->4.17, 4.2->4.17, 4.3->4.17, ...), but it only applies to one or two profiles, which is not that valuable. For 4.14->4.17 and 4.15->4.17 that have multiple profiles in it, we should be able to apply it to the 3 hops / 2 hops steps.

Reply:
Here we are talking the whole workload on implementing a dedicated step in chain upgrade, right? If so, both hops/steps number and profiles number will increase the workload. 
Or do u plan to split the chain upgrade into two styles, one is using the step as the implement in this pr, and another is keeping current setting in the yaml file. I'm afraid it's not easily understood for the maintainer.

Reply:
That looks like much more complicated than I have thought. 
Let's focus on 2 hops and 3 hops for cpou for now. We can discuss the chained upgrade in slack for a long term plan.